### PR TITLE
Split list formatter into smaller methods

### PR DESCRIPTION
**Pull request by Gerko32**
Split list formatter into smaller methods

This is a safe refactor before making more changes. My real goal is to change the links so they point to the right pilot in cases of same pilot name being for multiple ships (e.g. Anakin). I was first splitting this into smaller methods to understand the code

### CODE REVIEW

File: r2d7/listformatter.py [hunk 1]
```diff
@@ -114,21 +135,7 @@ def print_xws(self, xws, url=None):
...
-                    else:
-                        stat = 0
-                        for stat_block in pilot_card['ship']['stats']:
-                            if stat_block['type'] == cost['variable']:
-                                stat = stat_block['value']
-                                break
-                    points += cost['values'][str(stat)]
-                else:
-                    points += cost.get('value', 0)
+                points+=self.get_upgrade_cost(pilot_card, upgrade)
```
Comment (FreakyDug) [inline r2d7/listformatter.py:138]:
```suggestion
                points += self.get_upgrade_cost(pilot_card, upgrade)
```

**Comment by Gerko32** [comment]
Fixed as you requested.

For the wiki_link, it seems like naming has a pattern, so maybe we can have a dictionary. Where are you loading all the data? I I will give it a try in a separate merge request

**Comment by FreakyDug** [comment]
> 
> 
> Fixed as you requested.
> 
> For the wiki_link, it seems like naming has a pattern, so maybe we can have a dictionary. Where are you loading all the data? I I will give it a try in a separate merge request

The data is loaded here: https://github.com/FreakyDug/r2-d7/blob/8d5a0bc3fc26e85de264610c697892b6872d56b3/r2d7/core.py#L93
The cardlookup component has some hardcoded data structures similar to what you're describing here: https://github.com/FreakyDug/r2-d7/blob/8d5a0bc3fc26e85de264610c697892b6872d56b3/r2d7/cardlookup.py#L22 and does some preprocessing on the raw data here: https://github.com/FreakyDug/r2-d7/blob/8d5a0bc3fc26e85de264610c697892b6872d56b3/r2d7/cardlookup.py#L90

**Comment by danrs** [comment]
Hey @Gerko32, have you seen this post about the wiki? [https://community.fantasyflightgames.com/topic/310627-wiki-is-now-a-disaster-and-its-fandoms-fault/](https://community.fantasyflightgames.com/topic/310627-wiki-is-now-a-disaster-and-its-fandoms-fault/)

Fandom have rolled out a breaking change which has really messed up the x-wing Wiki, and Wazat hasn't decided yet whether he's going to update it to fix it.

I've never been a massive fan of the fandom wiki style anyway (especially on mobile), so I've been thinking about switching the R2-D7 links to point to [https://xhud.sirjorj.com/xwing.cgi/?](https://xhud.sirjorj.com/xwing.cgi/?) instead. This site is much cleaner and clearer, and also has the advantage of showing points history for cards. I've been chatting with sirjorj (the site author) and he's enabled an xws argument (`xws=`) which would be a lot less hassle than generating name-based links and matching them up with the wiki. What do you think?

**Comment by Gerko32** [comment]
Its sad to hear about fandom :( but I will love to help with this project. @danrs how could I help? do we know when will this new site start supporting xws? Is this site's code also in github?

**Comment by danrs** [comment]
@Gerko32 the site in question is [https://xhud.sirjorj.com/xwingcgi.html](https://xhud.sirjorj.com/xwingcgi.html). It is run by sirjorj, and the code is open source on gitlab at [https://gitlab.com/sirjorj/xwingcgi](https://gitlab.com/sirjorj/xwingcgi). 

In terms of how you can help, I've barely looked at the URL-generating code, so you probably have a better idea of what needs to be done than I do. My rough thinking was that we can probably just change the link-generating methods to build links to xwing.cgi instead of to the wiki, and along the way we can remove any of the nasty workarounds that were needed for wiki links

I've been chatting with sirjorj in the [X-Wing Vassal League Slack](https://join.slack.com/t/xwvl/shared_invite/zt-gulvc1he-~B5EH_2EY1GIoYgAt5t8Kg). He says the xws support is now online: you can use `xws=` as an arg instead of `pilot=`/`upgrade=`. For example: [https://xhud.sirjorj.com/xwing.cgi/pilot2?xws=nusquadronpilot](https://xhud.sirjorj.com/xwing.cgi/pilot2?xws=nusquadronpilot) or [https://xhud.sirjorj.com/xwing.cgi/upgrade2?xws=deadmansswitch](https://xhud.sirjorj.com/xwing.cgi/upgrade2?xws=deadmansswitch). It looks like a few things are still missing (currently it only supports pilots and upgrades, not ships or conditions etc) so I'm following up with him now in the #tech channel in that slack

**Comment by danrs** [comment]
sirjorj has now added support for xws on the `condition` and `ship` endpoints

**Pull request merged**